js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,147 @@
+//Task1
+function testValue (value) {
+  if (typeof value == 'number' || typeof value == 'object') {
+    return true;
+  };
+  if (typeof value == 'function') {
+    return false;
+  };
+  if (typeof value == 'string' && value.length != 10) {
+    return `Длина вашей строки: ${value.length}.`;
+  };
+  if (typeof value == 'string' && value.length == 10) {
+    return 'you win';
+  }
+}
+let typeofNumber = 10;
+let typeofObject = {
+  name: 'Natalia'
+};
+let typeofFunction = function () {
+  return 2 + 2;
+};
+let typeofString = 'JS';
+let typeofString10 = 'javascript';
+
+console.log(testValue(typeofNumber));//true
+console.log(testValue(typeofObject));//true
+console.log(testValue(typeofFunction));//false
+console.log(testValue(typeofString));//'длина вашей строки: <длина строки>'
+console.log(testValue(typeofString10));//'you win'
+
+
+
+//Task2
+function numbersBetween(a, b) {
+  if (typeof a == 'number' && typeof b == 'number') {
```

Comment:
that conditional nice, but I would be much nicer if you send some error message to a user.

In production code-base such code checks called "fast-fail"

and in most cases it looks like

if(typeof a =='number' && typeof b == 'number') {
 return
}
... another code comes here, without wrapping all code in if statment